### PR TITLE
fix: -Os should be used during compilation phase in makefile

### PR DESCRIPTION
**Pull request by theHamsta**
- also add `CXX_STANDARD` again as it's the only variable used from the Lua side

There might be more errors in the Makefile https://github.com/nvim-treesitter/nvim-treesitter/issues/2463#issue-1125016147 let's see what the problem was there. I would prefer to return back to the basic Makefile we had in the beginning (which already had mentioned error of using the optimization flag during linkage phase)

 https://github.com/nvim-treesitter/nvim-treesitter/issues/2463#issue-1125016147  was related to a `LDFLAGS` which was set in the user's environment and not by our Lua code.

### CODE REVIEW

**Comment by kyazdani42** [comment]
why would the code size optimization fix a linker issue :o ?

**Comment by theHamsta** [comment]
I only wanted to fix the `-Os` problem first, but the found that the other issue was indeed related to makefile variables set from the outside enviroment. I think it would be good to have the Makefile as simple as possible to avoid any more maintenance issues.

Best would be to have no alternative installation mechanism. But getting c++14 for norg, was really challenging any other way.

**Comment by theHamsta** [comment]
`-shared` should have been in LDFLAGS since LDFLAGS are sometimes understood as LD_EXTRAFLAGS it would be possible to set all variables in the makefile in Lua but I think that the general Makefile will find application out-side of nvim-treesitter while we only want to compile norg on Apple.

**Comment by theHamsta** [comment]
Fixes #2494

File: scripts/compile_parsers.makefile [hunk 1]
```diff
@@ -22,25 +18,21 @@ else
    rmf         = rm -rf $(1)
 endif
 
-ifneq ($(wildcard $(SRC_DIR)/*.cc),)
+ifneq ($(wildcard src/*.cc),)
```
Comment (kyazdani42) [inline scripts/compile_parsers.makefile:21]: is this change needed ?

Comment (kyazdani42) [reply scripts/compile_parsers.makefile:21]: actually all the removed substitutions ?

**Review by kyazdani42 — APPROVED** [review]

File: scripts/compile_parsers.makefile (hunk 1 above)
Comment (theHamsta) [reply scripts/compile_parsers.makefile]: Acutally, src was hard-coded in the targets anyways. `$(SRC_DIR)` might just be a risk that someone actually has this defined. but I changed it now to use `$(SRC_DIR)` which might be a bad idea
